drake/core/Function.h

```diff
@@ -104,46 +106,54 @@ namespace Drake {
     }
   };
 
-  class Function {
-  public:
-    Function() : relation(InputOutputRelation::Form::ARBITRARY) {};
-    Function(InputOutputRelation::Form f) : relation(f) {};
-    virtual ~Function() {};
+  template <typename ScalarType> using VecIn = Eigen::Ref< Eigen::Matrix<ScalarType, Eigen::Dynamic, 1> const >;
+  template <typename ScalarType> using VecOut = Eigen::Matrix<ScalarType, Eigen::Dynamic, 1>;
```

Comment:
Vector.h does not have an equivalent for either of these.  We could add them with names like `VectorRefConst<ScalarType>` and `DynamicVector<ScalarType>`.  However, those names are fairly verbose.  My thinking here is that part of the Function concept is to define the `eval()` method in terms of these type aliases for convenience, brevity, and clarity of argument purpose.  I used VecIn/VecOut as placeholders for now.  Perhaps we can choose better names.  This may be revised or clarified as part of the functional form work.